pkg/certificate/providers/certmanager/certificate_manager_test.go

```diff
@@ -285,3 +119,25 @@ func TestCertificateFromCertificateRequest(t *testing.T) {
 	assert.Nil(cert)
 	assert.Nil(err)
 }
+
+func TestNew(t *testing.T) {
+	assert := tassert.New(t)
+	fakeClient := cmfakeclient.NewSimpleClientset()
+	_, err := New(
+		&certificate.Certificate{},
+		fakeClient,
+		"osm-system",
+		cmmeta.ObjectReference{Name: "osm-ca"},
+		0,
+	)
+
+	assert.Error(err, "expected error from key size of zero, got")
```

Comment:
Nit only because this message is a little awkward when it does fail:
```
--- FAIL: TestNew (0.00s)
    certificate_manager_test.go:134:
                Error Trace:    certificate_manager_test.go:134
                Error:          An error is expected but got nil.
                Test:           TestNew
                Messages:       expected error from key size of zero, got
FAIL
FAIL    github.com/openservicemesh/osm/pkg/certificate/providers/certmanager    0.517s
FAIL
```

```suggestion
	assert.Error(err, "expected error from key size of zero")
```

Reply:
good call, fixed